pkg/kmm/preflightvalidationocp_test.go

```diff
@@ -0,0 +1,329 @@
+		},
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp object test doens't exist in namespace testns",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testPreflight := generatePreflight(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testPreflight)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		builderResult, err := PullPreflightValidationOCP(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, builderResult)
+			assert.Equal(t, testCase.name, builderResult.Definition.Name)
+			assert.Equal(t, testCase.namespace, builderResult.Definition.Namespace)
+		}
+	}
+}
+
+func TestPreflightCreate(t *testing.T) {
+	testCases := []struct {
+		addToRuntimeObjects bool
+	}{
+		{
+			addToRuntimeObjects: true,
+		},
+		{
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		testPreflight := generatePreflight(defaultPreflightName, defaultPreflightNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testPreflight)
+		}
+
+		testBuilder := buildPreflightTestBuilderWithFakeObjects(runtimeObjects)
+
+		builderResult, err := testBuilder.Create()
+		assert.Nil(t, err)
+		assert.NotNil(t, builderResult)
+		assert.Equal(t, defaultPreflightName, builderResult.Definition.Name)
+		assert.Equal(t, defaultPreflightNamespace, builderResult.Definition.Namespace)
+	}
+}
+func TestPreflightUpdate(t *testing.T) {
+
+}
+func TestPreflightExists(t *testing.T) {
+
+}
+func TestPreflightDelete(t *testing.T) {
+	testCases := []struct {
+		exists        bool
+		expectedError error
+	}{
+		{
+			exists:        true,
+			expectedError: nil,
+		},
+		{
+			exists:        false,
+			expectedError: errors.New("preflightvalidationocp cannot be deleted because it does not exist"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		testPreflight := generatePreflight(defaultPreflightName, defaultPreflightNamespace)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, testPreflight)
+		}
+
+		testBuilder := buildPreflightTestBuilderWithFakeObjects(runtimeObjects)
+
+		_, err := testBuilder.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testBuilder.Object)
+		}
+	}
+}
+
+func TestPreflightGet(t *testing.T) {
```

Comment:
Please refer to: pkg/argocd/argocd_test.go

Reply:
Updated. thanks